src/stream/mod.rs

```diff
@@ -148,33 +159,100 @@ impl Stream {
     /// This will download the video to <video_id>.mp4 in the current working directory.
     #[inline]
     pub async fn download(&self) -> Result<PathBuf> {
+        self.internal_download(None).await
+    }
+
+    /// Attempts to downloads the [`Stream`]s resource.
+    /// This will download the video to <video_id>.mp4 in the current working directory.
+    /// Takes an [`Callback`](crate::stream::callback::Callback)
+    #[cfg(any(feature = "callback", doc))]
+    #[doc(cfg(feature = "callback"))]
+    #[inline]
+    pub async fn download_callback(&self, callback: Callback) -> Result<PathBuf> {
+        self.internal_download(Some(callback)).await
+    }
+
+    #[inline]
+    async fn internal_download(&self, callback: Option<Callback>) -> Result<PathBuf> {
         let path = Path::new(self.video_details.video_id.as_str())
             .with_extension("mp4");
-        self.download_to(&path)
+        self.internal_download_to(&path, callback)
             .await
             .map(|_| path)
     }
 
     /// Attempts to downloads the [`Stream`]s resource.
-    /// This will download the video to <video_id>.mp4 in the provided directory. 
+    /// This will download the video to <video_id>.mp4 in the provided directory.
     #[inline]
     pub async fn download_to_dir<P: AsRef<Path>>(&self, dir: P) -> Result<PathBuf> {
+        self.internal_download_to_dir(dir, None).await
+    }
+
+    /// Attempts to downloads the [`Stream`]s resource.
+    /// This will download the video to <video_id>.mp4 in the provided directory. 
+    /// Takes an [`Callback`](crate::stream::callback::Callback)
+    #[cfg(any(feature = "callback", doc))]
+    #[doc(cfg(feature = "callback"))]
+    #[inline]
+    pub async fn download_to_dir_callback<P: AsRef<Path>>(
+        &self,
+        dir: P,
+        callback: Callback
+    ) -> Result<PathBuf> {
+        self.internal_download_to_dir(dir, Some(callback)).await
+    }
+
+    #[inline]
+    async fn internal_download_to_dir<P: AsRef<Path>>(
+        &self,
+        dir: P,
+        callback: Option<Callback>
+    ) -> Result<PathBuf> {
         let mut path = dir
             .as_ref()
             .join(self.video_details.video_id.as_str());
         path.set_extension("mp4");
-        self.download_to(&path)
+        self.internal_download_to(&path, callback)
             .await
             .map(|_| path)
     }
 
+    #[cfg(feature = "callback")]
```

Comment:
This feature gate breaks the `rustube-cli`. Removing it has no negative effects.

```
    Checking rustube-cli v0.2.4 (C:\Users\info\Code\Rust\rustube\cli)
error[E0599]: no method named `download_to` found for struct `rustube::Stream` in the current scope
   --> cli\src\bin\rustube.rs:191:18
    |
191 |                 .download_to(&path)
    |                  ^^^^^^^^^^^ help: there is an associated function with a similar name: `download`

error[E0599]: no method named `download_to` found for struct `rustube::Stream` in the current scope
   --> cli\src\bin\rustube.rs:202:18
    |
202 |                 .download_to(&filename)
    |                  ^^^^^^^^^^^ help: there is an associated function with a similar name: `download`
```

Reply:
Must have forgotten this one, fixed now